overlord/snapstate/snapmgr.go

```diff
@@ -94,6 +94,11 @@ type SnapSetup struct {
 	UserID  int    `json:"user-id,omitempty"`
 	Base    string `json:"base,omitempty"`
 
+	// Prereq is a list of snap-names that need to get installed
+	// together with this snap. Typically used when installing
+	// content-snaps with default-providers.
+	Prereq []string `json:"prereq,omitempty"`
```

Comment:
> This would be more like "Recommends", or perhaps "Wants". As discussed we also want to change the task name
> so it aligns with its purpose and with the usual conventions.
>
> This is probably best for a separate PR, though. Let's please catch up so we decide on good terms we're happy with.

Okay, here are suggestions for making the several names a bit more consistent, and to get this unblocked at least on that end.

I'm making two assumptions on those suggestions:

- That there's no implication of ordering on the installation of at least some of these snaps, in that a default provider, for example, is not required to be present upfront for the snap to work. For that reason these are not Pre-Requires, in the traditional package management sense.
- That these are indeed requirements. That is, the installation of the main snap will fail if anything on the list of required snaps fails to install before that.

If those two points hold, then we might go forward with these:

- Task: **"check-required-snaps"** (needs a compatibility handler for old one right below "cleanup", redirecting "prerequisites" to the same function)
- Handler: **doCheckRequiredSnaps**
- SnapSetup: **RequiredSnaps []string `json:"required-snaps"`**
- SnapManager func: **installRequiredSnaps**
- Parameters, variables, etc: **required**